typescript/sdk/src/deploy/middleware/LiquidityLayerRouterDeployer.ts

```diff
@@ -107,6 +139,65 @@ export class LiquidityLayerDeployer<
     };
   }
 
+  async deployPortalAdapter(
+    chain: Chain,
+    adapterConfig: PortalAdapterConfig,
+    owner: string,
+    router: LiquidityLayerRouter,
+  ): Promise<PortalAdapter> {
+    const cc = this.multiProvider.getChainConnection(chain);
+
+    const initCalldata =
+      PortalAdapter__factory.createInterface().encodeFunctionData(
+        'initialize',
+        [
+          ChainNameToDomainId[chain],
+          owner,
+          adapterConfig.portalBridgeAddress,
+          router.address,
+        ],
+      );
+    const portalAdapter = await this.deployContract(
+      chain,
+      'portalAdapter',
+      [],
+      {
+        create2Salt: this.create2salt,
+        initCalldata,
+      },
+    );
+
+    for (const {
+      wormholeDomain,
+      hyperlaneDomain,
+    } of adapterConfig.wormholeDomainMapping) {
+      const expectedCircleDomain =
+        await portalAdapter.hyperlaneDomainToWormholeDomain(hyperlaneDomain);
+      if (expectedCircleDomain === wormholeDomain) continue;
+
+      this.logger(
+        `Set wormhole domain ${wormholeDomain} for hyperlane domain ${hyperlaneDomain}`,
+      );
+      await cc.handleTx(
+        portalAdapter.addDomain(hyperlaneDomain, wormholeDomain),
+      );
+    }
+
+    if (
+      (await router.liquidityLayerAdapters('Portal')) !== portalAdapter.address
```

Comment:
I'd recommend comparing addresses using something that accounts for possible casing/formatting differences. Just checked, surprisingly we don't have one yet. Here a simple one I use in a few places:
```
import { getAddress } from '@ethersproject/address';

export function areAddressesEqual(a1: string, a2: string) {
  return getAddress(a1) === getAddress(a2);
}
```